velox/dwio/common/DataSink.cpp

```diff
@@ -28,6 +30,10 @@ FileSink::FileSink(
     const MetricsLogPtr& metricLogger,
     IoStatistics* stats)
     : DataSink{name, metricLogger, stats} {
+  auto dir = fs::path(name).parent_path();
```

Comment:
Shall we use Velox filesystem here?

Reply:
Hi @xiaoxmeng , I find DataSink abstraction and Velox FileSystem abstraction are kept separate now, ex., we register DataSink independently, https://github.com/facebookincubator/velox/blob/d38a666fc6e9e450aca1783828eaa6ff84a939a5/velox/dwio/common/DataSink.h#L200.
Current Velox has separate implementation of local FileSink and WarmStorageBlockSink, https://fburl.com/code/usn9qvb8. Each calls their corresponding "file system" APIs for file ops. I think we should follow this pattern.

As you might know, the reason we call Velox FileSystem interface is to auto dispatch to the right "file system" APIs based on the file path prefix. Here we're already with the local file DataSink, so we don't need the FileSystem interface for the Polymorphism

Reply:
Shall we use the same set of interface to access file system. My understanding is that local FileSink and  WarmStorageBlockSink could be a layer on top of file system abstraction. Ideally, we could have one DataSink object built on top of different file system interfaces?

Reply:
Can we use Velox filesystem interface to access local file system instead of directly calling posix APIs. The same for ws based data sink?